tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -243,12 +267,24 @@ class D3D12Device : public RendererBase
         {
         }
 
+        static uint32_t id;
         D3D12Resource m_resource;
         D3D12_RESOURCE_STATES m_defaultState;
 
-        virtual SLANG_NO_THROW Result SLANG_MCALL getNativeHandle(NativeHandle* outHandle) override
+        virtual SLANG_NO_THROW Result SLANG_MCALL getNativeResourceHandle(NativeResourceHandle* outHandle) override
+        {
+            outHandle->handle = (uint64_t)m_resource.getResource();
+            outHandle->type = NativeHandleType::D3D12;
+            return SLANG_OK;
+        }
+
+        virtual SLANG_NO_THROW Result SLANG_MCALL getSharedHandle(SharedHandle* outHandle) override
         {
-            *outHandle = (uint64_t)m_resource.getResource();
+            ComPtr<ID3D12Device> pDevice;
+            auto pResource = m_resource.getResource();
+            pResource->GetDevice(IID_PPV_ARGS(pDevice.writeRef()));
+            SLANG_RETURN_ON_FAIL(pDevice->CreateSharedHandle(pResource, NULL, GENERIC_ALL, (L"Texture" + std::to_wstring(id)).c_str(), outHandle));
```

Comment:
We need to have a policy on who is responsible for freeing the shared handle. My suggestion is that the Buffer/Texture resource itself is responsible for that.
To do that the BufferResource/TextureResource implementation need to have a member that stores the shared handle, and we call `CreateSharedHandle` only when that handle hasn't been created yet. If it has we just return the previously created handle without calling CreateSharedHandle again. We also need to make sure to call `CloseHandle` on the shared handle when the resource is destroyed.

Reply:
Added for D3D12 BufferResource, will gradually add more as the rest of the shared handle implementations come in